src/base/cache/cache_entry.hh

```diff
@@ -46,9 +59,25 @@ namespace gem5
 class CacheEntry : public ReplaceableEntry
 {
   public:
+    using IndexingPolicy = BaseIndexingPolicy;
+    using KeyType = Addr;
+    using TagExtractor = std::function<Addr(Addr)>;
+
     CacheEntry() = default;
```

Comment:
Can this class even work without a tag extractor? I think we may not need the register function, and instead we should always provide the extractor upon construction. By always registering the tag extractor here we guarantee that the classes that inherit from this one do not forget to register the extractor.

Reply:
Yes makes sense and I amended the PR accordingly. As a note: while this has been done for the CacheEntry, the TaggedEntry is still using the register method. The reason being is that some CacheBlk child classes are being allocated with the default constructor (example: https://github.com/gem5/gem5/blob/stable/src/mem/cache/tags/fa_lru.cc#L81)

They are anyway using their own tag extraction logic (https://github.com/gem5/gem5/blob/stable/src/mem/cache/tags/fa_lru.hh#L244) and not only that...

Reply:
FALRU always being the naysayer...